Add Uploader to `README.md` (#82)
cloudinary/cloudinary_kotlin: README.md

- [Usage](#usage)
    - [Setup](#Setup)
    - [Transform and Optimize Assets](#Transform-and-Optimize-Assets)

>     - [Uploading Asset](#Uploading-Assets)

Reviewer: ```suggestion
    - [Uploading Assets](#Uploading-Assets)
```

This will output the following url:
`https://res.cloudinary.com/demo/image/upload/c_fill,h_150,w_100/sample.jpg`

### Uploading Assets

The entry point for upload operations is the `cloudinary.uploader().upload()` call. All upload operations are dispatched to a background queue.

> The following examples uploads a `File` using the default settings, a request upload callback, and an upload preset (more about upload presets below):

Reviewer: ```suggestion
The following example performs an unsigned upload of a `File` using the default settings, a request upload callback, and an upload preset (required for unsigned uploads):
```

```kotlin
cloudinary.uploader().upload(imageFile) { 
    params {
      uploadPreset = "sample_preset"
    }
    options {

>       unsigned = true

Reviewer 2: isn't it a default?

Author: No, signed is default, if you to perform unsigned you need to specify it in the options object

    }
}
```

The uploaded image is assigned a randomly generated public ID, which is returned as part of the response object.
The image is immediately available for download through a CDN:

    cloudinary.image().generate("generatedPublicId")
      

>     http://res.cloudinary.com/demo/image/upload/generatedPublicId.jpg

Reviewer 2: maybe replace `demo` with `<your cloud name>` or something like that

Author: will do

        params {
          uploadPreset = "sample_preset"
          publicId = "sample_remote" 
        }
        options {
            unsigned = true
        }
}
```

For security reasons, mobile applications cannot contain the full account credentials, and so they cannot freely upload resources to the cloud.
Cloudinary provides two different mechanisms to enable end-users to upload resources without providing full credentials.

##### 1. Unsigned uploads using [Upload Presets.](https://cloudinary.com/documentation/upload_presets)
You can create an upload preset in your Cloudinary account console, defining rules that limit the formats, transformations, dimensions and more.
Once the preset is defined, it's name is supplied when calling upload. An upload call will only succeed if the preset name is used and the resource is within the preset's pre-defined limits.

The following example uploads a local resource, available as a Uri, assuming a preset named 'sample_preset' already exists in the account:

```kotlin
cloudinary.uploader().upload(uri) {
        params {
          uploadPreset = "sample_preset"
        }
        options {
          unsigned = true
        }
}
```

> ##### 2. Signed uploads with server-based signature

Reviewer: ```suggestion
##### 2. Signed uploads with server-generated signature
```

> Another way to allow uploading without credentials is using signed uploads.

Reviewer: ```suggestion
Another way to upload without including credentials is using signed uploads.
```

> It is recommended to generate the upload authentication signature on the server side, where it's safe to store the `api_secret`.

Reviewer: ```suggestion
You should generate the upload authentication signature on the server side, where it's safe to store the `api_secret`.
```

> Cloudinary's Kotlin SDK allows providing server-generated signature and any additional parameters that were generated on the server side (instead of signing using `api_secret` locally).

Reviewer: ```suggestion
The Cloudinary Kotlin SDK allows you to provide a server-generated signature and any additional parameters that were generated on the server side (instead of signing using `api_secret` locally).
```

> Your server can use any Cloudinary libraries (Ruby on Rails, PHP, Python & Django, Java, Perl, .Net, etc.) for generating the signature. The following JSON in an example of a response of an upload authorization request to your server:

Reviewer 2: I would provide a link to the documentation here (how to generate a signature)

Reviewer: Can use this link - https://cloudinary.com/documentation/upload_images#generating_authentication_signatures

Author: @njb90 should we replace the entire text with the link? do you want to give some context?
